src/libraries/System.Net.Mail/src/System/Net/Mail/MailAddress.cs

```diff
@@ -119,13 +119,18 @@ public static bool TryCreate(string address, string? displayName, Encoding? disp
 
         private static bool TryParse(string address, string? displayName, Encoding? displayNameEncoding, out (string displayName, string user, string host, Encoding displayNameEncoding) parsedData, bool throwExceptionIfFail)
         {
-            if (address == null)
+            if (string.IsNullOrEmpty(address))
             {
-                throw new ArgumentNullException(nameof(address));
-            }
-            if (address.Length == 0)
-            {
-                throw new ArgumentException(SR.Format(SR.net_emptystringcall, nameof(address)), nameof(address));
+                if (throwExceptionIfFail)
+                {
+                    if (address is null)
+                        throw new ArgumentNullException(nameof(address));
+                    else
+                        throw new ArgumentException(SR.Format(SR.net_emptystringcall, nameof(address)), nameof(address));
```

Comment:
Please either do:
```suggestion
                    if (address is null)
                    {
                        throw new ArgumentNullException(nameof(address));
                    }
                    else
                    {
                        throw new ArgumentException(SR.Format(SR.net_emptystringcall, nameof(address)), nameof(address));
                    }
```
or
```suggestion
                    throw address is null ?
                        new ArgumentNullException(nameof(address)) :
                        new ArgumentException(SR.Format(SR.net_emptystringcall, nameof(address)), nameof(address));
```

Reply:
Done!